kani-compiler/src/codegen_cprover_gotoc/codegen/intrinsic.rs

```diff
@@ -1361,6 +1361,22 @@ impl<'tcx> GotocCtx<'tcx> {
         )
     }
 
+    /// Generates code for a SIMD vector comparison intrinsic. This is similar
+    /// to `binop`, but we pass along `ret_ty` because return type for vector
+    /// comparison depends on the place type.
+    fn codegen_intrinsic_simd_cmp<F: FnOnce(Expr, Expr, Type) -> Expr>(
```

Comment:
I think we should type check the expression and print user friendly errors, like the `llvm` backend does. 
- Check that return type length matches the rhs operands
- Check that the return type base type is an integer.

https://github.com/rust-lang/rust/blob/c1a859b25a95999ba276075bbd8e284ea675b41a/compiler/rustc_codegen_llvm/src/intrinsic.rs#L925-L953

Reply:
Done. The errors look like this now:
```
error: expected return type with length 2 (same as input type `Vector { typ: Unsignedbv { width: 64 }, size: 2 }`), found `Vector { typ: Unsignedbv { width: 32 }, size: 4 }` with length 4
  --> main.rs:37:35
   |
37 |         let invalid_simd: u32x4 = simd_eq(x, y);
   |                                   ^^^^^^^^^^^^^

error: aborting due to previous error
```

Reply:
Nice! Any chance you can print the rust types instead of the `goto` ones? I.e.: print `u64x2` instead of `Vector { typ: Unsignedbv { width: 64 }, size: 2 }`

Reply:
Done!